Netcracker/KubeMarine
# [CPDEV-89888] Support for plugins resources requests/limits

### Description
Currently kubemarine OOB plugins are installed without resources requests/limits, this is not recommended, we need to set some defaults and allow users to change them.

### Solution
* Added default requests/limits in `defaults.yaml` for each OOB plugin container (calico has only requests by default)
* Updated schema so that user can override default values in `cluster.yaml`
* If user provides requests/limits manually, defaults are not applied for particular container
* Added enrichment of resulting requests/limits to plugins manifest
* Added inventory patch, which sets resources requests/limits to backward-compatible values (mostly empty)

### How to apply
* Make sure OOB plugins resources requests/limits are configured as needed in `cluster.yaml`
* `deploy.plugins`

### Test Cases

1. Install the cluster using KubeMarine without these changes (e.g. previous version). Run `kubereq`, it should show errors and warnings about missing cpu and memory requets/limits for OOB plugins
2. Take KubeMarine with these changes. Run `kubemarine_migrate --force-apply plugins_resources`. Your `cluster.yaml` should be updated with `resources` for each OOB plugin container. Run `deploy.plugins` and make sure `kubereq` shows exactly the same errors/warnings as on step 1. This is to ensure that patch keeps backward compatibility.
3. Remove all `resources` sections from your plugins in `cluster.yaml`, which were added by patch. Run `deploy.plugins` and make sure `kubereq` no longer show requests/limits errors/warning as before. This is to ensure that default resources are applied and satisfy our recommendations. Make sure plugins are running normally with default resources.
4. Specify your own `resources` for some or all OOB plugins containers. Run `deploy.plugins` and make sure plugins pods have resources exactly as you specified, without "additional defaults". This is to make sure our defaults are not mixed with user-provided values.

### Checklist
